### PR TITLE
Feat: Add Wishlist management

### PR DESCRIPTION
**Pull request by starver20**
[Netlify Preview](https://deploy-preview-7--stupefied-swanson-69d789.netlify.app/)
Added wishlist management for frontend
Wishlist is an array on ids of products which are wishlisted.
It is compared with products array to find which product is wishlisted and make the heart logo filled with red color for it

### CODE REVIEW

**Review by mohammed-sas — APPROVED** [review]
1.the code is structured well
2. suggestion : my preference would be to keep wishlist and cart context seperate so that debugging would be easier, also we would be sending only the needed data not both cart and wishlist items.

**Comment by starver20** [comment]
> 1.the code is structured well 2. suggestion : my preference would be to keep wishlist and cart context seperate so that debugging would be easier, also we would be sending only the needed data not both cart and wishlist items.

Thank you. I had previously kept different context for wishlist and cart context, but then to update the wishlist , i had to send 2 different dispatch requests, so i decided to keep them together

File: src/components/card/CartCard.jsx [hunk 1]
```diff
@@ -68,18 +81,18 @@ const CartCard = (product) => {
           </button>
         </div>
         <p className={classes['prod-price']}>₹{price}</p>
-        <span className={classes.fav}>
+        <span onClick={wishlistClickHandler} className={classes.fav}>
```
Comment (VishalPatil18) [inline src/components/card/CartCard.jsx:84]: You could use one type of convention for using classes. Like the one above using [...] would be better. The point being just use any one kind.
E.g.
`className = {classes['fav']}`

**Review by VishalPatil18 — APPROVED** [review]
Just one suggestion added. The rest looks quite good. Great work @starver20

File: src/components/card/CartCard.jsx (hunk 1 above)
Comment (starver20) [reply src/components/card/CartCard.jsx:84]: All right ill keep it in mind, thanks.

**Pull request merged**